tests/api/test_authentication.py

```diff
@@ -3,82 +3,68 @@
 from sqlalchemy.orm import Session
 
 from ensysmod import crud
-from tests.utils.utils import random_lower_string
+from tests.utils.data_generator.users import generate_credentials, new_user, user_create_request
+from tests.utils.utils import assert_response
 
 
-def get_register_payload():
-    return {"username": random_lower_string(), "password": random_lower_string()}
-
-
-def test_register_endpoint(client: TestClient, db: Session):
-    payload = get_register_payload()
-    r = client.post("/auth/register", json=payload)
-    new_user = r.json()
-    assert r.status_code == status.HTTP_200_OK
-    assert new_user
-    assert new_user['username'] == payload['username']
-    db_user = crud.user.get_by_username(db=db, username=payload['username'])
-    assert db_user
+def test_register_endpoint(client: TestClient):
+    create_request = user_create_request()
+    response = client.post("/auth/register", content=create_request.model_dump_json())
```

Comment:
```suggestion
    response = client.post("/auth/register", json=create_request.model_dump_json())
```
wouldn' t this be the better choice here?